src/Nethermind/Nethermind.JsonRpc/Modules/Subscribe/ISubscriptionFactory.cs

```diff
@@ -0,0 +1,26 @@
+//  Copyright (c) 2021 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using Nethermind.JsonRpc.Modules.Eth;
+
+namespace Nethermind.JsonRpc.Modules.Subscribe
+{
+    public interface ISubscriptionFactory
+    {
+        Subscription CreateSubscription(SubscriptionType subscriptionType, Filter filter);
```

Comment:
I am not complaining here, just laughing at our previous problems with subscriptions
this is yet another subscription factory in out system (there are probably four of them now)